src/gui/search/searchwidget.cpp

```diff
@@ -162,6 +163,15 @@ bool SearchWidget::eventFilter(QObject *object, QEvent *event)
             closeTab(tabIndex);
             return true;
         }
+        if (mouseEvent->button() == Qt::RightButton)
+        {
+            QMenu *menu = new QMenu(this);
+            menu->setAttribute(Qt::WA_DeleteOnClose);
+            menu->addAction(tr("Close tab"), this, [this, tabIndex]() { closeTab(tabIndex); });
+            menu->addAction(tr("Close all tabs"), this, &SearchWidget::closeAllTabs );
```

Comment:
Now you have redundant space before `)`.

Reply:
is there any way for me to check this by some software?

Reply:
Theoretically, yes. If you can set it up correctly.
I don't trust anything but my eyes. This is not so difficult when you are used to seeing properly formatted code.
Of course, I use some autoformatting features to filter out some errors, such as extra trailing spaces.

Reply:
You have incorrect formatting even in human language text (commit message). You should insert space after punctuation marks like `. , ! ?` etc. `.` is missing in the end of `Closes #5628` sentence.
So I wouldn't expect any software to fix all your mistakes.

Reply:
okay, but it will help a lot)